### ISSUE TITLE
Test suite segfault on Windows 10 with Ruby 2.3.1

### ISSUE THREAD

**Issue by djberg96** [djberg96/win32-taskscheduler#12]
Windows 10
ruby 2.3.1p112 (2016-04-26 revision 54768) [x64-mingw32]
ffi 1.9.18 x64-mingw32

Just running `rake test` results in the following:

```
...C:/Users/djber/Dev/win32-taskscheduler/lib/win32/taskscheduler.rb:628: [BUG] Segmentation fault
ruby 2.3.1p112 (2016-04-26 revision 54768) [x64-mingw32]

-- Control frame information -----------------------------------------------
c:0023 p:---- s:0115 e:000114 CFUNC  :method_missing
c:0022 p:0090 s:0110 e:000109 METHOD C:/Users/djber/Dev/win32-taskscheduler/lib/win32/taskscheduler.rb:628
c:0021 p:0016 s:0096 e:000095 METHOD C:/Users/djber/Dev/win32-taskscheduler/test/test_taskscheduler.rb:39
c:0020 p:0008 s:0093 e:000092 LAMBDA C:/Users/djber/Dev/win32-taskscheduler/test/test_taskscheduler.rb:52 [FINISH]
c:0019 p:0083 s:0091 e:000090 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testcase.rb:718
c:0018 p:0064 s:0087 e:000086 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testcase.rb:458
c:0017 p:0029 s:0083 e:000082 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testsuite.rb:124
c:0016 p:0097 s:0075 e:000074 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testsuite.rb:53
c:0015 p:0029 s:0069 E:001288 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testsuite.rb:124
c:0014 p:0097 s:0061 e:000060 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testsuite.rb:53
c:0013 p:0035 s:0055 E:001238 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnermediator.rb:65
c:0012 p:0050 s:0051 e:000050 BLOCK  C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnermediator.rb:44
c:0011 p:0084 s:0049 E:001d08 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnermediator.rb:100
c:0010 p:0069 s:0041 E:000ee8 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnermediator.rb:40
c:0009 p:0011 s:0035 e:000034 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunner.rb:40
c:0008 p:0033 s:0032 e:000031 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunner.rb:25
c:0007 p:0021 s:0029 e:000028 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnerutilities.rb:24
c:0006 p:0015 s:0024 e:000023 BLOCK  C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/autorunner.rb:411
c:0005 p:0033 s:0022 e:000021 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/autorunner.rb:467
c:0004 p:0174 s:0018 E:001910 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/autorunner.rb:410
c:0003 p:0079 s:0013 e:000012 METHOD C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/autorunner.rb:59
c:0002 p:0047 s:0005 e:000003 BLOCK  C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit.rb:502 [FINISH]
c:0001 p:0000 s:0002 E:000ce0 (none) [FINISH]

-- Ruby level backtrace information ----------------------------------------
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit.rb:502:in `block (2 levels) in <top (required)>'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/autorunner.rb:59:in `run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/autorunner.rb:410:in `run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/autorunner.rb:467:in `change_work_directory'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/autorunner.rb:411:in `block in run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnerutilities.rb:24:in `run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunner.rb:25:in `start'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunner.rb:40:in `start_mediator'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnermediator.rb:40:in `run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnermediator.rb:100:in `with_listener'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnermediator.rb:44:in `block in run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/ui/testrunnermediator.rb:65:in `run_suite'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testsuite.rb:53:in `run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testsuite.rb:124:in `run_test'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testsuite.rb:53:in `run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testsuite.rb:124:in `run_test'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testcase.rb:458:in `run'
C:/tools/ruby23/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/testcase.rb:718:in `run_test'
C:/Users/djber/Dev/win32-taskscheduler/test/test_taskscheduler.rb:52:in `block in <class:TC_TaskScheduler>'
C:/Users/djber/Dev/win32-taskscheduler/test/test_taskscheduler.rb:39:in `setup_task'
C:/Users/djber/Dev/win32-taskscheduler/lib/win32/taskscheduler.rb:628:in `new_work_item'
C:/Users/djber/Dev/win32-taskscheduler/lib/win32/taskscheduler.rb:628:in `method_missing'

-- C level backtrace information -------------------------------------------
C:\Windows\SYSTEM32\ntdll.dll(NtWaitForSingleObject+0x14) [0x00007FF85D576154]
C:\Windows\System32\KERNELBASE.dll(WaitForSingleObjectEx+0x8f) [0x00007FF859BA75FF]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_bugreport+0xa4) [0x000000006E6AE944]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_bug_context+0x68) [0x000000006E54B7C8]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_check_safe_obj+0x3b2) [0x000000006E61DF52]
 [0x0000000000401A85]
C:\Windows\SYSTEM32\ntdll.dll(_C_specific_handler+0x96) [0x00007FF85D566BD6]
C:\Windows\SYSTEM32\ntdll.dll(_chkstk+0x11d) [0x00007FF85D57AB9D]
C:\Windows\SYSTEM32\ntdll.dll(RtlImageNtHeaderEx+0x483) [0x00007FF85D519913]
C:\Windows\SYSTEM32\ntdll.dll(KiUserExceptionDispatcher+0x3a) [0x00007FF85D579CBA]
 [0x0000000065282CF3]
 [0x0000000065283FA6]
 [0x0000000065285A1A]
 [0x0000000065288E13]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_error_arity+0x1fb) [0x000000006E69743B]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_invoke_proc+0x760) [0x000000006E6A38A0]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_invoke_proc+0xdee) [0x000000006E6A3F2E]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_invoke_proc+0x7ea) [0x000000006E6A392A]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0xe03) [0x000000006E69DF83]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0x57d8) [0x000000006E6A2958]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0x5b54) [0x000000006E6A2CD4]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0x5ce4) [0x000000006E6A2E64]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_invoke_proc+0x118) [0x000000006E6A3258]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_invoke_proc+0x760) [0x000000006E6A38A0]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_invoke_proc+0xa30) [0x000000006E6A3B70]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0xe03) [0x000000006E69DF83]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0x57d8) [0x000000006E6A2958]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0x5e7f) [0x000000006E6A2FFF]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_localjump_error+0x5f91) [0x000000006E6A3111]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_vm_invoke_proc+0x2a) [0x000000006E6A316A]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_proc_call+0x44) [0x000000006E559DA4]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_exec_end_proc+0x1e2) [0x000000006E552FF2]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(rb_exec_end_proc+0x2c5) [0x000000006E5530D5]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(ruby_cleanup+0x30a) [0x000000006E5536BA]
C:\tools\ruby23\bin\x64-msvcrt-ruby230.dll(ruby_run_node+0x58) [0x000000006E553968]
 [0x0000000000402D14]
 [0x00000000004013D7]
 [0x00000000004014F8]
C:\Windows\System32\KERNEL32.DLL(BaseThreadInitThunk+0x14) [0x00007FF85CDF8364]
```

**Comment by djberg96** [comment]
Note that I was able to run the tests without issue using a 32-bit binary.

**Comment by djberg96** [comment]
Something about setting the description and/or author to an empty string seems to be killing it.

**Comment by phasis68** [comment]
I cannot reproduce this issue on Windows 10
ruby 2.3.3p222 (2016-11-21 revision 56859) [x64-mingw32]

It looks like WIN32OLE related issue.

**Comment by djberg96** [comment]
@phasis68 Yes, it looks like whatever the issue was it was corrected in 2.3.3. I'll add a note to the README about potential issues. Thanks.
